### ISSUE TITLE
"flux jobs -A -f pending,running" lists jobs whose states are neither pending nor running

### ISSUE THREAD

**Issue by ofaaland** [flux-framework/flux-core#3993]
[root@elmerfudd1:~]# which flux
/bin/flux

[root@elmerfudd1:~]# rpm -qf /bin/flux
flux-core-0.30.0-1.t4.x86_64

[root@elmerfudd1:~]# flux jobs -A -f pending,running | head
       JOBID USER     NAME       ST NTASKS NNODES  RUNTIME NODELIST
 ?A7nDuT6HFm faaland1 flux        R      2      1    1.37m elmerfudd15
 ?A7k3EEmWPZ faaland1 flux       CD      2      1   4.642m elmerfudd15
 ?A78U4ZGT4s faaland1 flux       CD      1      1   0.129s elmerfudd15
 ?A78AwqfMWo faaland1 flux       CD      1      1   33.29s elmerfudd15
 ?A786dALnq5 faaland1 flux       CD      1      1   0.152s elmerfudd15
 ?A77dkxPdrP faaland1 flux       CD      1      1   56.23s elmerfudd15
 ?A73df7Yz9u faaland1 flux       CD      1      1   8.715m elmerfudd15
 ?6FZfjVWbYj faaland1 flux       CD      1      1   14.01m elmerfudd15
 ?6FZj7RDgbq faaland1 flux       CD      1      1   0.052s elmerfudd15

**Comment by ofaaland** [comment]
It may be that the man page is just out of date.  It says:
```
       -f, --filter=STATE|RESULT
              List jobs with specific job state or result. Multiple states or  results  can  be
              listed  separated  by  comma.  See  JOB  STATUS below for additional information.
              Defaults to pending,running.
```

which led me to believe that a comma-delimited list of states was valid input.  However, it doesn't actually say that anywhere, nor does the output of ```flux jobs -h```.

**Comment by ofaaland** [comment]
Actually, it's not just the comma-delimited list of states that is at issue.  state "running" also doesn't seem to work.
```
[root@elmerfudd1:~]# flux jobs -A -f running | head
       JOBID USER     NAME       ST NTASKS NNODES  RUNTIME NODELIST
 ?A7nDuT6HFm faaland1 flux        R      2      1    12.7m elmerfudd15
 ?A7k3EEmWPZ faaland1 flux       CD      2      1   4.642m elmerfudd15
 ?A78U4ZGT4s faaland1 flux       CD      1      1   0.129s elmerfudd15
 ?A78AwqfMWo faaland1 flux       CD      1      1   33.29s elmerfudd15
 ?A786dALnq5 faaland1 flux       CD      1      1   0.152s elmerfudd15
 ?A77dkxPdrP faaland1 flux       CD      1      1   56.23s elmerfudd15
 ?A73df7Yz9u faaland1 flux       CD      1      1   8.715m elmerfudd15
 ?6FZfjVWbYj faaland1 flux       CD      1      1   14.01m elmerfudd15
 ?6FZj7RDgbq faaland1 flux       CD      1      1   0.052s elmerfudd15
```

**Comment by grondo** [comment]
This should be fixed. The issue is because `-A` lists "all jobs for all users" including jobs that have completed, therefore the `--filter` argument is overridden.
I've even run across this myself and been frustrated, so we should fix it.

For now, I think `flux jobs --user=all -f running` is probably what you want. Sorry for the confusion.

(and my guess is that this will be a common use case, so we should add a single option that does the equivalent)

**Comment by ofaaland** [comment]
Thanks @grondo that is indeed what I want.  I somehow overlooked that ```-A``` didn't just mean ```--user=all``` even though the man page is clear about that.

**Comment by grondo** [comment]
I have to agree it is confusing. Maybe `-A` _should_  just be shorthand for `--user=all`. In any event, I think `--user=all -f running` and probably also `--user=all -f running,pending` will probably be common queries and we'll want an easy way to do that, so let's leave this issue open until we have something like that implemented.

**Comment by chu11** [comment]
looking at the manpage really quick

```
       -a     List all jobs of the current user, including inactive jobs.  Equivalent to specifying --filter=pending,running,inactive.

       -A     List all jobs from all users, including inactive jobs. Equivalent to specifying --filter=pending,running,inactive --user=all.
```

I like the idea that perhaps `-a` is shorthand for `--filter=pending,running,active` (re-word text to something like "list jobs in all states") and `-A` is shorthand for `--user=all`.  So doing `flux-jobs -a -A` could be a somewhat common usage.

And a `flux-jobs -A -f pending,running` would totally work.

**Comment by grondo** [comment]
@chu11, was this one closed by #4012?

**Comment by chu11** [comment]
doh, I guess I didn't add a "Fixes" into the commit message since  that PR was initially a WIP / discussion if this would be an acceptable fix.  Will close.
